op-challenger/config/config.go

```diff
@@ -17,79 +17,38 @@ import (
 	txmgr "github.com/ethereum-optimism/optimism/op-service/txmgr"
 )
 
-var (
-	ErrMissingL1EthRPC       = errors.New("missing l1 eth rpc url")
-	ErrMissingRollupRpc      = errors.New("missing rollup rpc url")
-	ErrMissingL2OOAddress    = errors.New("missing l2 output oracle contract address")
-	ErrMissingDGFAddress     = errors.New("missing dispute game factory contract address")
-	ErrInvalidNetworkTimeout = errors.New("invalid network timeout")
-	ErrMissingTxMgrConfig    = errors.New("missing tx manager config")
-	ErrMissingRPCConfig      = errors.New("missing rpc config")
-	ErrMissingLogConfig      = errors.New("missing log config")
-	ErrMissingMetricsConfig  = errors.New("missing metrics config")
-	ErrMissingPprofConfig    = errors.New("missing pprof config")
-)
-
 // Config is a well typed config that is parsed from the CLI params.
 // This also contains config options for auxiliary services.
 // It is used to initialize the challenger.
 type Config struct {
 	// L1EthRpc is the HTTP provider URL for L1.
 	L1EthRpc string
 
-	// RollupRpc is the HTTP provider URL for the rollup node.
-	RollupRpc string
-
-	// L2OOAddress is the L2OutputOracle contract address.
-	L2OOAddress common.Address
+	// GameAddress is the DisputeGameFactory contract address.
+	GameAddress common.Address
 
-	// DGFAddress is the DisputeGameFactory contract address.
-	DGFAddress common.Address
+	// Alphabet Trace is a temporary trace provider
+	AlphabetTrace string
 
 	// NetworkTimeout is the timeout for network requests.
 	NetworkTimeout time.Duration
 
-	TxMgrConfig *txmgr.CLIConfig
-
-	RPCConfig *oprpc.CLIConfig
-
-	LogConfig *oplog.CLIConfig
-
-	MetricsConfig *opmetrics.CLIConfig
-
-	PprofConfig *oppprof.CLIConfig
+	TxMgrConfig   txmgr.CLIConfig
+	RPCConfig     oprpc.CLIConfig
+	LogConfig     oplog.CLIConfig
+	MetricsConfig opmetrics.CLIConfig
+	PprofConfig   oppprof.CLIConfig
 }
 
 func (c Config) Check() error {
 	if c.L1EthRpc == "" {
-		return ErrMissingL1EthRPC
-	}
-	if c.RollupRpc == "" {
-		return ErrMissingRollupRpc
-	}
-	if c.L2OOAddress == (common.Address{}) {
-		return ErrMissingL2OOAddress
-	}
-	if c.DGFAddress == (common.Address{}) {
-		return ErrMissingDGFAddress
-	}
-	if c.NetworkTimeout == 0 {
-		return ErrInvalidNetworkTimeout
-	}
-	if c.TxMgrConfig == nil {
-		return ErrMissingTxMgrConfig
+		return errors.New("empty l1 rpc")
 	}
-	if c.RPCConfig == nil {
-		return ErrMissingRPCConfig
+	if c.GameAddress == (common.Address{}) {
+		return errors.New("empty game address")
```

Comment:
If we keep these errors defined, we can keep some config tests which I believe are still somewhat useful and follow `op-program` testing precedent.

Reply:
Restored these errors & the tests. Tests did change b/c the configs really should have been structs, not pointers.